eth/vm/logic/flow.py

```diff
@@ -57,3 +58,36 @@ def gas(computation: BaseComputation) -> None:
     gas_remaining = computation.get_gas_remaining()
 
     computation.stack_push_int(gas_remaining)
+
+
+def beginsub(computation: BaseComputation) -> None:
+    #TODO: raise OOG exception
+    pass
+
+
+def jumpsub(computation: BaseComputation) -> None:
+    sub_loc = computation.stack_pop1_int()
+
+    temp = computation.code.program_counter
+    computation.code.program_counter = sub_loc
+
+    next_opcode = computation.code.peek()
+
+    if next_opcode != BEGINSUB:
+        #TODO: abort execution
+        pass
+    
+    else:
+        computation.code.program_counter += 1
+    
+    computation.rstack_push_int(temp + 1)
+
+
+def returnsub(computation: BaseComputation) -> None:
+
+    if computation.rstack.length == 0:
```

Comment:
rather than doing this check explicitely I think it would more closely follow our existing conventions to have the call to `computation.rstack_pop1_int()` throw an exception and to do this via a `try/except` block.